MetaMask/metamask-extension
# feat: adds "data collection for marketing" toggles

Adds data collection for marketing toggles (and toasts/warnings) on:
- Onboarding
- Toast in Wallet
- Settings page



## **Description**



[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/24605?quickstart=1)

## **Related issues**

Fixes:
https://github.com/MetaMask/MetaMask-planning/issues/2437
https://github.com/MetaMask/MetaMask-planning/issues/2438
https://github.com/MetaMask/MetaMask-planning/issues/2526

## **Manual testing steps**

Onboarding checkbox:
Make the `metametrics.js` to return `renderOnboarding` instead of `renderLegacyOnboarding`
1. Start a new account
2. There should be a new checkbox that asks for marketing consent
3. Checking it should set the marketing consent to true (check at Settings, Securty tab page)

Security tab:
1. Go to Security tab
2. When checking the "Data collection for marketing" to `true`, the "Participate in MetaMetrics" toggle should turn to `true`
3. When checking "Participate in MetaMetrics" to `false`, "Data collection for marketing" should be set to `false`
4. When "Participate in Metametrics" is `true` and "Data collection for marketing" is `true`, and the latter is set to `false`, a warning message should appear.

Toast:
An already onboarded user will have the "dataCollectionForMarketing" value as `null` (neither `true` or `false`). This will trigger the toast.
1. By clicking on "I accept", it should set the "Data collection for marketing" to `true`.
2. By closing the toast or clicking on "No thanks", it should set the "Data collection for marketing" to `false`.

All of these actions should trigger subsequent Segment events.


## **Screenshots/Recordings**



### **Before**



### **After**

<img width="646" alt="Screenshot 2024-05-20 at 14 19 53" src="https://github.com/MetaMask/metamask-extension/assets/11148144/a13b70ec-1d89-4e9f-8df1-2096dd56e642">

https://github.com/MetaMask/metamask-extension/assets/11148144/5eb59902-768d-4d07-a112-5aeb5471587d



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
